### DISCUSSION TITLE
Can't install mitmproxy on M1 using pip

### DISCUSSION THREAD

**Discussion by tomerb-neosec** [mitmproxy/mitmproxy#6731] [Q&A]
I'm running a docker file on my M1, in this dockerfile i'm trying to install mitmproxy.
The installation runs forever.
This is the last line which it is stuck on: Building wheel for mitmproxy-rs (pyproject.toml) ...

NOTE: I ran the same docker build in my intel macbook and it succeed.

This is the docker image i'm using: owasp/zap2docker-bare:b2024-03-04

**Comment by mhils** [comment]
We provided precompiled wheels for macOS that work both on x86_64 as well as arm64: https://pypi.org/project/mitmproxy-rs/#files. It should typically pick up those instead of compiling its own thing. My suspicion would be that you are running an outdated Python version.
